framework/JobHandler.py

```diff
@@ -161,128 +160,110 @@ def __checkAndRemoveFinished(self, running):
         self.raiseAMessage(" Process Failed " + str(running) + " internal returnCode " + str(returnCode))
         self.__failedJobs[running.identifier]=(returnCode,copy.deepcopy(metadataToKeep))
 
-  def __initializeParallelPython(self):
+  def __initializeRay(self):
     """
       Internal method that is aimed to initialize the internal parallel system.
-      It initilizes the paralle python implementation (with socketing system) in
+      It initilizes the RAY implementation (with socketing system) in
       case RAVEN is run in a cluster with multiple nodes or the NumMPI > 1,
       otherwise multi-threading is used.
       @ In, None
       @ Out, None
     """
+    ## set up enviroment
+    os.environ['PYTHONPATH'] = os.pathsep.join(sys.path)
     ## Check if the list of unique nodes is present and, in case, initialize the
-    ## socket
+    servers = None
     if self.runInfoDict['internalParallel']:
       if len(self.runInfoDict['Nodes']) > 0:
         availableNodes = [nodeId.strip() for nodeId in self.runInfoDict['Nodes']]
-
-        ## Set the initial port randomly among the user accessible ones
-        ## Is there any problem if we select the same port as something else?
-        randomPort = random.randint(1024,65535)
-
+        ## identify the local host name and get the number of local processors
+        localHostName = self.__getLocalHost()
+        self.raiseADebug("Local host name is  : "+ localHostName)
+        nProcsHead = availableNodes.count(localHostName)
+        self.raiseADebug("# of local procs    : "+ str(nProcsHead))
+        ## initialize ray server with nProcs
+        self.rayserver = ray.init(num_cpus=int(nProcsHead))
```

Comment:
1. camelBack for rayserver?
2. It seems to me this variable should be defined in the init first.

Reply:
done